src/components/AppBar.tsx

```diff
@@ -260,7 +262,11 @@ class AppBar extends React.Component<Props, State> {
                           color="inherit"
                           aria-label="menu"
                           className={classes.menuButton}
-                          onClick={(): void => this.props.history.goBack()}
+                          onClick={async (): void => {
+                            if(await preBack()) {
```

Comment:
This is the conditional I was talking about.

Reply:
That conditional is receiving the value passed into the resolve method. I think I _was_ wrapping the promise I wanted in one I didn't need, but then resolving it twice, so it did eventually work, just not how I was thinking. My most recent commit gets rid of that outer promise.